pkg/server/endpoints/endpoints.go

```diff
@@ -202,7 +203,7 @@ func (e *Endpoints) createUDSServer(unaryInterceptor grpc.UnaryServerInterceptor
 	return grpc.NewServer(
 		grpc.UnaryInterceptor(unaryInterceptor),
 		grpc.StreamInterceptor(streamInterceptor),
-		grpc.Creds(auth.UntrackedUDSCredentials()))
+		grpc.Creds(peertracker.NewCredentials()))
```

Comment:
Hmm I wonder if there are compatibility concerns here. If a SPIRE Server is running in a k8s pod without `hostPid` set, is the CLI still usable?

Reply:
that is exactly what worry me, and why audit log must be optional, 
but I didnt found another options to get IDs, 
What do you think about allowing different levels of audit logs?
- Disable: no audit log is provided (default)
- Enable: enable audit log with full capabilities
- Restricted: add audit log but disable UDS tracking 

With restricted feauture we can allow adding audit logs with TLS tracking but not with UDS tracking, in that way we allow some audit features for cases where allowing `hostPid` is not an option

Reply:
I made this comment already in a different place in the PR, but for posterity, we settled on only allowing the first two options @MarcosDY mentioned - either disabled (default behavior) or fully enabled for both TCP and UDS endpoints, but no "half-enabled" mode where TCP endpoint auditing is enabled, but UDS endpoint auditing is not.